xcdat/temporal.py

```diff
@@ -958,16 +994,32 @@ def _set_arg_attrs(
 
         # "season" frequency specific configuration attributes.
         for key in season_config.keys():
-            if key not in DEFAULT_SEASON_CONFIG.keys():
+            # TODO: Deprecate `drop_incomplete_djf`.
+            valid_keys = list(DEFAULT_SEASON_CONFIG.keys()) + ["drop_incomplete_djf"]
+            if key not in valid_keys:
                 raise KeyError(
                     f"'{key}' is not a supported season config. Supported "
                     f"configs include: {DEFAULT_SEASON_CONFIG.keys()}."
                 )
         custom_seasons = season_config.get("custom_seasons", None)
         dec_mode = season_config.get("dec_mode", "DJF")
-        drop_incomplete_djf = season_config.get("drop_incomplete_djf", False)
 
         self._season_config: SeasonConfigAttr = {}
+
+        # TODO: Deprecate `drop_incomplete_djf`.
+        drop_incomplete_djf = season_config.get("drop_incomplete_djf", None)
+        if drop_incomplete_djf is not None:
+            warnings.warn(
+                "The `season_config` argument 'drop_incomplete_djf' is being "
+                "deprecated. Please use 'drop_incomplete_seasons' instead.",
+                DeprecationWarning,
+            )
```

Comment:
TODO: Need to a specify a specific version that we will deprecate `drop_incomplete_djf`. Probably v0.8.0.